dev_docs/07_visual_testing_guide.md

```diff
@@ -0,0 +1,93 @@
+# Visual Testing
+
+The KDS now has a visual testing system integrated into it, using which you can verify the expected visual states after making any changes to the components.
+
+## Implementation details
+
+  The visual testing mechanism uses Puppeteer(for interacting with the testing environment and the rendered components), Jest-Puppeteer(to provide all required configuration to run tests using Puppeteer.), and Percy(to take snapshots for comparing visual diffs) to ensure components render correctly under various conditions. The key components include:
```

Comment:
To improve readability. This could be reframed to something like:
```
The visual testing mechanism uses the following dependencies to ensure components render correctly under various conditions:
* Puppeteer: for interacting with the testing environment and the rendered components.
* Jest-Puppeteer: to provide all required configuration to run tests using Puppeteer.
* Percy: to take snapshots for comparing visual diffs.
```